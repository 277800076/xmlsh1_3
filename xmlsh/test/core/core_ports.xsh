# Test ports
unset doc doc2 doc3 xdoc
# Input XML ports
xread xdoc < ../../samples/data/books.xml
echo There are $(xpath 'count(//ITEM)' <{xdoc}) ITEMS

# Text Input , Text output, external comma 
doc=$(<../../samples/data/books.csv)
cat <{doc} >{doc2}

# Text input xml output
csv2xml <{doc2} >{doc3}

xquery '/root/row[2]/col[3]/string()' <{doc3}

xquery -i $doc3 '/root/row[2]/col[3]/string()'

unset doc doc2 doc3 xdoc

# Test sequences stored into variables
xpath -n "1,2,'foo',3" >{_seq}
echo <[ $_seq[3] ]>

xquery -n "<foo/>,<bar><spam>bletch</spam></bar>,3" >{_seq}
echo <[ $_seq[2]//spam ]>



