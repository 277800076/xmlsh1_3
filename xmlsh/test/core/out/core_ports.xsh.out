There are 6 ITEMS
Modern Library
Modern Library
foo
There are 3 elements
<spam>bletch</spam>
There are 4 elements
<spam>bletch</spam>
foobar should be foobar
foobar <foo/>
